rust_src/src/ffi.rs

```diff
@@ -0,0 +1,57 @@
+//! Module that is used for FFI exports.
+/// Used for creating #[no_mangle] wrappers around functions defined in the top level rust
+/// crate for remacs. These calls should NOT be used in Rust directly,
+/// it is only for C compat.
+use data;
+use keyboard;
+use lisp::LispObject;
+use lists;
+use math;
+use remacs_sys::{Lisp_Object, Lisp_Window};
```

Comment:
You did not follow the pattern.

std
external crates
remacs other crates
local files

Reply:
Thanks for the reminder, I had forgotten we adopted that convention. Will update.

Reply:
Follow up: `cargo fmt` is insisting I order the crates in the way they are currently ordered. Moving remacs_sys to the top of the list triggers `cargo fmt` to complain.

Reply:
Look at the other files. As long as you put a space between the groups rustfmt will honor the grouping.

Reply:
ah, thats what I was missing